server/src/main/java/org/opensearch/rest/BytesRestResponse.java

```diff
@@ -107,22 +110,30 @@ public BytesRestResponse(RestChannel channel, Exception e) throws IOException {
 
     public BytesRestResponse(RestChannel channel, RestStatus status, Exception e) throws IOException {
         ToXContent.Params params = paramsFromRequest(channel.request());
+        boolean emptyIndex = checkForEmptyIndex(channel.request().params());
```

Comment:
I don't think this is quite the right place to be doing the empty index check because this class doesn't know anything about the specific operation that has happened so again there is risk that there are some operations where an empty index is okay and this would be a breaking change. Ideally the parameter validation should happen on the request path in a class that understands the operation being performed and can make assertions about what constitutes valid parameters for that operation. I'd suggest finding the exact spot where the index out of bounds error occurs and see if you can add the validation at or before that point on the request path. Another really good practice in a case like this is to start by writing a test that fails and then go implement the fix.

Reply:
Also, as Tianli helpfully pointed out above, it may not have anything to do with an empty index parameter name, but the error message does suggest that somewhere we are attempting to do something like `String.substring` on an empty string or otherwise failing while attempting to parse a string. If you can isolate the exact point where that is happening then it will probably be much easier to figure out the appropriate fix.